### ISSUE TITLE
Request: Android API level 24

### ISSUE THREAD

**Issue by kingonlive** [anggrayudi/android-hidden-api#13]
Many thanks!! When can we get N version android.jar?

**Comment by anggrayudi** [comment]
We must wait until the full name is released. After that, we need to find a real device (not _emulator_) with API 24 (not _preview version_) to get `framework.jar` located in `/system/framework` directory. Then, upload this jar to me, so that I can make it available for API level 24.

I'll keep this issue open until someone upload the jar. Don't pick `framework.jar` from preview version of Android N!

**Comment by Splder** [comment]
https://www.dropbox.com/s/cuae130vro3ejdm/framework.jar

**Comment by anggrayudi** [comment]
@Splder from which device you get the `framework.jar`?

**Comment by Splder** [comment]
@anggrayudi this was taken from a Nexus 6P running PureNexus

So far this is the only framework.jar that i have been able to find that have the classes.dex files included

http://forum.xda-developers.com/android/apps-games/app-swipe-navigation-navigation-bar-t3272478/page12

I also have boot.oat file for aosp nexue 5X if you like

I was able to convert the boot.oat to odex but i'm having trouble de optimizing the dex

**Comment by anggrayudi** [comment]
Currently, dex files in the framework.jar you sent wont converted to jar file. I got this exception:
```
com.googlecode.d2j.DexException: not support version.
    at com.googlecode.d2j.reader.DexFileReader.<init>(DexFileReader.java:151)
    at com.googlecode.d2j.reader.DexFileReader.<init>(DexFileReader.java:211)
    at com.googlecode.d2j.tools.Dex2jarCmd.doCommandLine(Dex2jarCmd.java:104)
    at com.googlecode.d2j.tools.BaseCmd.doMain(BaseCmd.java:254)
    at com.googlecode.d2j.tools.Dex2jarCmd.main(Dex2jarCmd.java:32)
```

The previous version of jars were converted successfully. I don't know why this happen. Seems that type of the dex files in API 24 are different with earlier version.

**Comment by Splder** [comment]
@anggrayudi i also get the same problem maybe we need the framework from a deodex ROM?

**Comment by anggrayudi** [comment]
Why don't we try it?

**Comment by abhinaykukkadapu** [comment]
@anggrayudi how are you so 100% sure that it is not supported may be we have to deodex in another way?

**Comment by oginski** [comment]
Whole directory framework from physical device LG G5:

http://www110.zippyshare.com/v/2QxEoftQ/file.html

**Comment by zhangsl** [comment]
Why not compile a jar from AOSP directly?

**Comment by DVDAndroid** [comment]
@anggrayudi You can use [this dex2jar](https://github.com/anggrayudi/android-hidden-api/files/657744/dex-tools-2.1-SNAPSHOT.zip) to decompile `framework.jar` for API 24

**Comment by wajdidhc** [comment]
any progress on this

**Comment by anggrayudi** [comment]
We just released Hidden API for Nougat 7.0 (API 24).
